### PR TITLE
Make hot reload work w/ rn

### PR DESCRIPTION
**Pull request by chrisnojima**
alrite, so this kinda sucks...

I went down the path of getting the packager to work with symlinks but its just a big PITA:
https://github.com/facebook/react-native/issues/637#issuecomment-201320363

Instead i made react-native/shared not be a symlink but instead be a place we just sync our files to. Not ideal but this really simplifies things. We could apply this to desktop also if we wanted.... Symlinks, while it worked, makes things hard (esp. on windows). A nice side effect of this is that the syncing filters out the desktop files.

I'll update the docs about this if we move forward with this PR

So how this works now:
- You must have matchman installed ```brew install watchman``` (the npm version is old and broken)
- In react-native, do ```npm run watch```

Now when things change in client/shared it'll rsync them to react-native/shared, which'll trigger an RN packager build

@keybase/react-hackers feedback pleez

### CODE REVIEW

File: react-native/.gitignore [hunk 1]
```diff
@@ -22,6 +22,7 @@ npm-debug.log
 project.xcworkspace
 xcuserdata
 Pods
+shared
```
Comment (chrisnojima) [inline react-native/.gitignore]: this isn't checked in now

File: react-native/package.json [hunk 2]
```diff
@@ -20,7 +22,8 @@
     "moment": "2.12.0",
     "purepack": "keybase/purepack#nojima/keybase-client-changes",
     "qrcode-generator": "1.0.0",
-    "react-native": "0.21.0",
+    "react": "0.14.7",
```
Comment (chrisnojima) [inline react-native/package.json]: react is a peer dependency and is needed

File: react-native/rn-cli.config.js [hunk 3]
```diff
@@ -1,20 +0,0 @@
-const config = require('react-native/packager/rn-cli.config')
```
Comment (chrisnojima) [inline react-native/rn-cli.config.js]: we don't need this anymore now

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by cjb** [comment]
Tested on Linux, seems to do the right thing (though I had to compile watchman from source), :thumbsup:

**Comment by chrisnojima** [comment]
@MarcoPolo you ok with this

**Comment by MarcoPolo** [comment]
Can we update the README with instructions?

**Comment by chrisnojima** [comment]
yes i wanted to make sure we're fine with this first

**Comment by chrisnojima** [comment]
updated

**Comment by MarcoPolo** [comment]
:+1:

**Pull request merged**